src/osd/OSD.cc

```diff
@@ -4436,6 +4436,10 @@ void OSD::build_initial_pg_history(
       h->last_epoch_split = e;
     }
     lastmap = osdmap;
+    up_primary = new_up_primary;
```

Comment:
Actually we could update these fields only if a new_interval is detected https://github.com/ceph/ceph/pull/17423/files#diff-fa6c2eba8356ae1442d1bf749beacfdfL4424

Reply:
agreed. see #17437